mmf/models/transformers/base.py

```diff
@@ -53,24 +45,62 @@ class BaseTransformerConfigType:
     finetune_lr_multiplier: float  # finetune lr multiplier for base transformer
 
 
+class BaseTransformerBackend(nn.Module, ABC):
+    def __init__(self, config: BaseTransformerConfigType, *args, **kwargs):
+        super().__init__()
+        self.config = config
+        self.build_transformer_config()
+        self.build_transformer_base()
+        self.build_embeddings()
+
+    @abstractmethod
+    def build_transformer_config(self):
+        """Build the transformer base model config.
+
+        Warning: Empty shell for code to be implemented in other class.
+        """
+
+    @abstractmethod
+    def build_transformer_base(self):
+        """Build the transformer base model.
+
+        Warning: Empty shell for code to be implemented in other class.
+        """
+
+    @abstractmethod
+    def build_embeddings(self):
+        """Build the multimodal embeddings using the transformer base
+        embeddings.
+
+        Warning: Empty shell for code to be implemented in other class.
+        """
+
+    @abstractmethod
+    def get_config(self):
+        """Return the transformer configuration. This can be the config built
+        in `build_transformer_config` or the model config passed to init.
+
+        Warning: Empty shell for code to be implemented in other class.
+        """
```

Comment:
raise NotImplementedError in these.

Reply:
Since these are abstract methods, will complain anyway if not implemented.